From 6d985d29d8be2b70ee7e4048a7ae95e2b4013f17 Mon Sep 17 00:00:00 2001
From: Michael Orlitzky <michael@orlitzky.com>
Date: Fri, 28 Dec 2018 10:01:56 -0500
Subject: [PATCH 4/4] Ignore "SANITIZED ... NULL byte(s)" messages.

Newer versions of the amavisd daemon sanitize NULL bytes within the
body of a message for the benefit of Cyrus IMAP. When it does this,
the number of bytes sanitized is logged. This is an informational
message only as far as I can tell, so it's been added to the list of
messages to ignore.
---
 amavis-logwatch | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/amavis-logwatch b/amavis-logwatch
index 250a6ba..044c94d 100644
--- a/amavis-logwatch
+++ b/amavis-logwatch
@@ -2052,6 +2052,12 @@ sub create_ignore_list() {
    # describe what is being sent to the systemd notification socket,
    # if one exists.
    push @ignore_list_final, qr/^sd_notify( \(no socket\))?:/;
+
+   # In amavisd-new-2.11.0-rc1 and later, amavis will replace any null
+   # bytes that it finds in the body of a message with a "modified
+   # UTF-8" encoded null. The number of times it does this is then
+   # logged with the following message.
+   push @ignore_list_final, qr/^smtp forwarding: SANITIZED (\d+) NULL byte\(s\)/;
 }
 
 # Notes:
-- 
2.19.2

